### PR TITLE
Scrape flights: complete airlines, airplanes, airports, routes, shared routes

### PR DESCRIPTION
**Pull request by paulld**

### CODE REVIEW

**Comment by harrychenca** [comment]
All done ready to review?

**Comment by paulld** [comment]
yes it's ready for review. it's only backend though

File: app/controllers/api/v1/air_routes_controller.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
+module Api
+  module V1
+    class AirRoutesController < ApplicationController
+
+      def index
+        if params[:airline]
+          airline = Airline.find_by(:icao_iata => params[:airline])
+          @air_routes = airline.air_routes.all
+        else
+          @air_routes = AirRoute.all.limit(100)
```
Comment (harrychenca) [inline app/controllers/api/v1/air_routes_controller.rb]: Why still render when there's no `params[:airline]`

File: db/migrate/20150224084611_create_air_routes.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
+class CreateAirRoutes < ActiveRecord::Migration
+  def change
+    create_table :air_routes do |t|
+
+      t.string :code    # IATA ??
+      t.integer :airline_id
+      t.integer :air_shared_routes_count
+      t.integer :air_flights_count
+
```
Comment (harrychenca) [inline db/migrate/20150224084611_create_air_routes.rb]: Did you miss `from_airport_id` and `to_airport_id`?

File: db/migrate/20150224090530_create_air_flights.rb [hunk 3]
```diff
@@ -0,0 +1,36 @@
+class CreateAirFlights < ActiveRecord::Migration
```
Comment (harrychenca) [inline db/migrate/20150224090530_create_air_flights.rb]: No need to include `air_route_id`?

In `air_flight.rb`,

it says
```ruby
class AirFlight < ActiveRecord::Base

  belongs_to :airplane
  belongs_to :air_route, :counter_cache => true
end
```

File: db/migrate/20150224090530_create_air_flights.rb [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+  def change
+    create_table :air_flights do |t|
+
+      t.integer :airline_id
+      t.integer :airplane_id
+      t.integer :from_airport_id
+      t.integer :to_airport_id
```
Comment (harrychenca) [inline db/migrate/20150224090530_create_air_flights.rb]: Do you need `from_airport_id` and  `to_airport_id` here?

File: db/migrate/20150224090530_create_air_flights.rb [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+      t.integer :route_id
```
Comment (harrychenca) [inline db/migrate/20150224090530_create_air_flights.rb]: Why do we need `t.integer :route_id`?

File: app/models/airport.rb [hunk 6]
```diff
@@ -0,0 +1,11 @@
+class Airport < ActiveRecord::Base
+
+  validates :iata, presence: true, uniqueness: true
+
+  # has_many :from_airports, through: :air_flights_as_from_airports, class_name: "Airport"
+  # has_many :to_airports, through: :air_flights_as_to_airports, class_name: "Airport"
+  # has_many :air_flights_as_from_airports, class_name: "Flight", foreign_key: "from_airport_id", dependent: :destroy
+  # has_many :air_flights_as_to_airports,   class_name: "Flight", foreign_key: "to_airport_id",   dependent: :destroy
+  has_many :air_routes_as_from_airports, class_name: "Flight", foreign_key: "from_airport_id", dependent: :destroy
+  has_many :air_routes_as_to_airports,   class_name: "Flight", foreign_key: "to_airport_id",   dependent: :destroy
```
Comment (harrychenca) [inline app/models/airport.rb]: Or should the class_name be `AirRoute` instead of `Flight`?

**Comment by harrychenca** [comment]
Took some time to take a look at the schema. Everything looks okay, but got some questions.

File: app/models/airport.rb (hunk 6 above)
Comment (paulld) [inline app/models/airport.rb]: Indeed, I'll change that

File: db/migrate/20150224090530_create_air_flights.rb (hunk 5 above)
Comment (paulld) [inline db/migrate/20150224090530_create_air_flights.rb]: flight belongs_to route (for instance AF188 is the Paris > Hong Kong route operated by Air France, and every day there is a new flight taking that route)

**Comment by paulld** [comment]
Thanks, I'll make a few changes
* route indeed needs from_airport_id and to_airport_id, but flight doesn't
* By scraping the routes and shared_routes I have get their codes, not their respective from_airport_id and to_airport_id yet
* haven't scraped the flights yet

**Comment by harrychenca** [comment]
I am going to take another look. The structure is a bit complex. Need to sit down and go through it 1 by 1.

File: db/migrate/20150224090530_create_air_flights.rb (hunk 5 above)
Comment (harrychenca) [inline db/migrate/20150224090530_create_air_flights.rb]: should be `air_route_id`

File: db/migrate/20150224090530_create_air_flights.rb (hunk 4 above)
Comment (harrychenca) [inline db/migrate/20150224090530_create_air_flights.rb]: No need

File: db/migrate/20150224090530_create_air_flights.rb (hunk 3 above)
Comment (harrychenca) [inline db/migrate/20150224090530_create_air_flights.rb]: Solved. should change `route_id` to `air_route_id`

File: db/migrate/20150224084611_create_air_routes.rb (hunk 2 above)
Comment (harrychenca) [inline db/migrate/20150224084611_create_air_routes.rb]: Yes, need them

**Comment by paulld** [comment]
Issues are fixed. After merge & push to Heroku I'll handle the migrations with `rake db:migrate:dow VERSION=` and `rake db:migrate:up VERSION=`

**Pull request merged**